azext_iot/common/shared.py

```diff
@@ -304,5 +309,43 @@ class SHAHashVersions(Enum):
     """
     Supported SHA types for generating the certificate thumbprint.
     """
+
     SHA1 = 1
     SHA256 = 256
+
+
+class CertInfo(TypedDict):
```

Comment:
We currently support Python 3.7 ...TypedDict was introduced in Python 3.8

Reply:
one option is to say we only support 3.8 now

Reply:
Disagree, we should deprecate python versions when they aren't supported anymore. Python 3.7 EOL is in 8 months.

Reply:
The beauty of TypedDict is that we can just replace it with `Dict[str, str]` or `Dict[str, Any]` - it was just a convenience type for the certificate object format we pass around. I'll get rid of the TypedDict for now.

Reply:
Makes sense. I'm all for using the best facilities of Python we can, but we have to balance with compatibility and casting the widest (safe/secure) net of usage. We saw offline there are **still** notable usage of legacy/out-of-support Py versions.

Reply:
Yup - totally agree. I actually didn't see your comment when I typed mine, so I was just reiterating support for waiting to use it until we're able to ensure compatibility. If anything, it was worth using during dev for the intellisense alone 😆